Sources/ApodiniDeployBuildSupport/DeployedSystem.swift

```diff
@@ -37,74 +59,112 @@ public struct DeployedSystem: Codable {
         self.userInfo = try JSONEncoder().encode(userInfo)
         try nodes.assertHandlersLimitedToSingleNode()
     }
-    
-    
-    public func readUserInfo<T: Decodable>(as _: T.Type) -> T? {
-        try? T(decodingJSON: userInfo)
-    }
+//    
+//    
+//    public func readUserInfo<T: Decodable>(as _: T.Type) -> T? {
+//        try? T(decodingJSON: userInfo)
+//    }
 }
 
 
-extension DeployedSystem {
+extension AnyDeployedSystem {
     /// Fetch one of the system's nodes, by id.
-    public func node(withId nodeId: Node.ID) -> Node? {
+    public func node(withId nodeId: DeployedSystemNode.ID) -> DeployedSystemNode? {
         nodes.first { $0.id == nodeId }
     }
     
     /// Returns the node which exports an endpoint with the specified handler identifier.
     /// - Note: A system should never contain multiple nodes exporting the same endpoint,
-    public func nodeExportingEndpoint(withHandlerId handlerId: AnyHandlerIdentifier) -> Node? {
+    public func nodeExportingEndpoint(withHandlerId handlerId: AnyHandlerIdentifier) -> DeployedSystemNode? {
         nodes.first { $0.exportedEndpoints.contains { $0.handlerId == handlerId } }
     }
 }
 
+/// A node within the deployed system
+public struct DeployedSystemNode: Codable, Identifiable, Hashable, Equatable {
+    /// ID of this node
+    public let id: String
+    /// exported handler ids
+    public let exportedEndpoints: Set<ExportedEndpoint>
+    /// Additional deployment provider specific data
+    public private(set) var userInfo: Data?
```

Comment:
Maybe it makes sense to make this generic over a specific `UserInfo` type that is Codable?